tests/resources/mlflow-test-plugin/mlflow_test_plugin/dummy_evaluator.py

```diff
@@ -42,65 +44,94 @@ def _log_metrics(self, run_id, metrics, dataset_name):
-            )
-            confusion_matrix_image_artifact._save(confusion_matrix_image_artifact_name + ".png")
-            client.log_image(
-                run_id, confusion_matrix_image, confusion_matrix_image_artifact_name + ".png"
-            )
+                confusion_matrix_image_artifact_name = (
+                    f"confusion_matrix_image_on_{self.dataset.name}"
+                )
+                confusion_matrix_image_artifact = ImageEvaluationArtifact(
+                    uri=get_artifact_uri(
+                        self.run_id, confusion_matrix_image_artifact_name + ".png"
+                    ),
+                    content=confusion_matrix_image,
+                )
+                confusion_matrix_image_artifact._save(confusion_matrix_image_artifact_name + ".png")
+                self.client.log_image(
+                    self.run_id,
+                    confusion_matrix_image,
+                    confusion_matrix_image_artifact_name + ".png",
+                )
 
-            artifacts = {
-                confusion_matrix_artifact_name: confusion_matrix_artifact,
-                confusion_matrix_image_artifact_name: confusion_matrix_image_artifact,
-            }
-        elif model_type == "regressor":
-            mean_absolute_error = sk_metrics.mean_absolute_error(y, y_pred)
-            mean_squared_error = sk_metrics.mean_squared_error(y, y_pred)
+                artifacts = {
+                    confusion_matrix_artifact_name: confusion_matrix_artifact,
+                    confusion_matrix_image_artifact_name: confusion_matrix_image_artifact,
+                }
+        elif self.model_type == "regressor":
+            mean_absolute_error = sk_metrics.mean_absolute_error(self.y, y_pred)
+            mean_squared_error = sk_metrics.mean_squared_error(self.y, y_pred)
             metrics = {
                 "mean_absolute_error": mean_absolute_error,
                 "mean_squared_error": mean_squared_error,
             }
-            self._log_metrics(run_id, metrics, dataset.name)
+            if not self.is_baseline_model:
+                self._log_metrics(self.run_id, metrics, self.dataset.name)
             artifacts = {}
         else:
-            raise ValueError(f"Unsupported model type {model_type}")
+            raise ValueError(f"Unsupported model type {self.model_type}")
 
         return EvaluationResult(metrics=metrics, artifacts=artifacts)
+
+    # pylint: disable=unused-argument
+    def evaluate(
+        self, *, model, model_type, dataset, run_id, evaluator_config, baseline_model=None, **kwargs
+    ):
+        self.model_type = model_type
+        self.client = MlflowClient()
+        self.dataset = dataset
+        self.run_id = run_id
+        self.X = dataset.features_data
+        self.y = dataset.labels_data
+        y_pred = model.predict(self.X)
+        eval_result = self._evaluate(
+            y_pred, is_baseline_model=evaluator_config.get("is_baseline_model", False)
```

Comment:
I don't see a test where you set `is_baseline_model` for the the `evaluator_config` for the Dummy Evaluator. Do you need this `evaluator_config.get("is_baseline_model", False)`?

Reply:
Set to false